python/cuml/neighbors/nearest_neighbors_mg.pyx

```diff
@@ -0,0 +1,350 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+from cuml.neighbors import NearestNeighbors
+
+import numpy as np
+import pandas as pd
+import cudf
+import ctypes
+import cuml
+import warnings
+
+from cuml.common.base import Base
+from cuml.utils import get_cudf_column_ptr, get_dev_array_ptr, \
+    input_to_dev_array, zeros, row_matrix
+
+from cython.operator cimport dereference as deref
+
+from cuml.common.handle cimport cumlHandle
+from cuml.common.vector cimport vector
+
+import scipy.stats as stats
+
+
+from libcpp cimport bool
+from libcpp.memory cimport shared_ptr
+
+import rmm
+from libc.stdlib cimport malloc, free
+
+from libc.stdint cimport uintptr_t, int64_t
+from libc.stdlib cimport calloc, malloc, free
+
+from numba import cuda
+import rmm
+
+
+cimport cuml.common.handle
+cimport cuml.common.cuda
+
+
+cdef extern from "cumlprims/opg/matrix/data.hpp" namespace \
+        "MLCommon::Matrix":
+
+    cdef cppclass Data[T]:
+        Data(T *ptr, size_t totalSize)
+
+    cdef cppclass floatData_t:
+        floatData_t(float *ptr, size_t totalSize)
+        float *ptr
+        size_t totalSize
+
+ctypedef Data[int64_t] int64Data_t
+
+
+cdef extern from "cumlprims/opg/matrix/part_descriptor.hpp" namespace \
+        "MLCommon::Matrix":
+
+    cdef cppclass RankSizePair:
+        int rank
+        size_t size
+
+    cdef cppclass PartDescriptor:
+        PartDescriptor(size_t M,
+                       size_t N,
+                       vector[RankSizePair*] &partsToRanks,
+                       int myrank)
+
+cdef extern from "cumlprims/opg/selection/knn.hpp" namespace \
+        "MLCommon::Selection::opg":
+
+    cdef void brute_force_knn(
+        cumlHandle &handle,
+        vector[int64Data_t*] &out_I,
+        vector[floatData_t*] &out_D,
+        vector[floatData_t*] &idx_data,
+        PartDescriptor &idx_desc,
+        vector[floatData_t*] &query_data,
+        PartDescriptor &query_desc,
+        bool rowMajorIndex,
+        bool rowMajorQuery,
+        int k,
+        size_t batch_size,
+        bool verbose
+    ) except +
+
+
+class NearestNeighborsMG(NearestNeighbors):
+    """
+    Multi-node multi-GPU Nearest Neighbors kneighbors query.
+
+    NOTE: This implementation of NearestNeighbors is meant to be
+    used with an initialized cumlCommunicator instance inside an
+    existing distributed system. Refer to the Dask NearestNeighbors
+     implementation in `cuml.dask.neighbors.nearest_neighbors`.
+    """
+    def __init__(self, batch_size=1<<21, **kwargs):
+        super(NearestNeighborsMG, self).__init__(**kwargs)
+        self.batch_size = batch_size
+
+    def _build_dataFloat(self, arr_interfaces):
```

Comment:
Hmm. I normally use camel case, but also use _'s to separate the names into different components based on their meanings. In this case, the `build` is an action I'm performing on `dataFloat`. Otherwise, it's `_build_data_float`, which seems to separate the `data` and `float` so they lose their meaning as an instance of `floatData_t`. Really, I think this should be renamed to `build_floatData` (I believe I renamed the underlying `floatData_t` from `dataFloat_t` at one point. 

I used `_index_partsToRanks` for the same reason- since it's representing a c++ argument named `partsToRanks`.  Is this a bad practice in Cython/Python?